File: leetcode/solutions/0599.MinimumIndexSumOfTwoLists/findRestaurant_test.go
package solution

import (
	"sort"
	"testing"

	"github.com/google/go-cmp/cmp"
)

func Test_0599_MinimumIndexSumOfTwoLists(t *testing.T) {
	t.Parallel()

	type in = struct {
		arg1 []string
		arg2 []string
	}
	tests := []struct {
		in  in
		out []string
	}{
		{
			in{
				[]string{"Shogun", "KFC", "Burger King"},
				[]string{"KFC", "Shogun", "Burger King"},
			},
			[]string{"Shogun", "KFC"},
		},
		{
			in{
				[]string{"Shogun", "Tapioca Express", "Burger King", "KFC"},
				[]string{"Piatti", "The Grill at Torrey Pines", "Hungry Hunter Steakhouse", "Shogun"},
			},
			[]string{"Shogun"},
		},
		{
			in{
				[]string{"Shogun", "Tapioca Express", "Burger King", "KFC"},
				[]string{"KFC", "Shogun", "Burger King"},
			},
			[]string{"Shogun"},
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := findRestaurant(tt.in.arg1, tt.in.arg2)
			sort.Strings(got)
			sort.Strings(tt.out)

			if !cmp.Equal(got, tt.out) {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/0599.MinimumIndexSumOfTwoLists/findRestaurant_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import findRestaurant from "./findRestaurant.ts";

Deno.test("0599. Minimum Index Sum of Two Lists", () => {
  type test = [[string[], string[]], string[]];
  const tests: test[] = [
    [
      [
        ["Shogun", "KFC", "Burger King"],
        ["KFC", "Shogun", "Burger King"],
      ],
      ["Shogun", "KFC"],
    ],
    [
      [
        ["Shogun", "Tapioca Express", "Burger King", "KFC"],
        [
          "Piatti",
          "The Grill at Torrey Pines",
          "Hungry Hunter Steakhouse",
          "Shogun",
        ],
      ],
      ["Shogun"],
    ],
    [
      [
        ["Shogun", "Tapioca Express", "Burger King", "KFC"],
        ["KFC", "Shogun", "Burger King"],
      ],
      ["Shogun"],
    ],
    [
      [
        ["Shogun", "Tapioca Express", "Burger King", "KFC"],
        ["Tapioca Express", "Burger King", "Shogun"],
      ],
      ["Tapioca Express"],
    ],
  ];

  const sorter = (a: string, b: string) => b.localeCompare(a);

  for (const [args, want] of tests) {
    assertEquals(findRestaurant(...args).sort(sorter), want.sort(sorter));
  }
});
